modules/e2ee/OlmAdapter.js

```diff
@@ -470,6 +603,266 @@ export class OlmAdapter extends Listenable {
             }
             break;
         }
+        case OLM_MESSAGE_TYPES.SAS_START: {
+            if (!olmData.session) {
+                logger.debug(`Received sas init message from ${pId} but we have no session for them!`);
+
+                this._sendError(participant, 'No session found while processing sas-init');
+
+                return;
+            }
+
+            if (olmData.sasVerification?.sas) {
+                logger.warn(`SAS already created for participant ${pId}`);
+                this.eventEmitter.emit(
+                    OlmAdapterEvents.PARTICIPANT_VERIFICATION_COMPLETED,
+                    pId,
+                    false,
+                    E2EEErrors.E2EE_SAS_INVALID_SAS_VERIFICATION);
+
+                return;
+            }
+
+            const { transactionId } = msg.data;
+
+            const sas = new Olm.SAS();
+
+            olmData.sasVerification = {
+                sas,
+                transactionId,
+                isInitiator: false
+            };
+
+            const pubKey = olmData.sasVerification.sas.get_pubkey();
+            const commitment = this._computeCommitment(pubKey, msg.data);
+
+            /* The first phase of the verification process, the Key agreement phase
+                https://spec.matrix.org/latest/client-server-api/#short-authentication-string-sas-verification
+            */
+            const acceptMessage = {
+                [JITSI_MEET_MUC_TYPE]: OLM_MESSAGE_TYPE,
+                olm: {
+                    type: OLM_MESSAGE_TYPES.SAS_ACCEPT,
+                    data: {
+                        transactionId,
+                        commitment
+                    }
+                }
+            };
+
+            this._sendMessage(acceptMessage, pId);
+            break;
+        }
+        case OLM_MESSAGE_TYPES.SAS_ACCEPT: {
+            if (!olmData.session) {
+                logger.debug(`Received sas accept message from ${pId} but we have no session for them!`);
+
+                this._sendError(participant, 'No session found while processing sas-accept');
+
+                return;
+            }
+
+            const { commitment, transactionId } = msg.data;
+
+
+            if (!olmData.sasVerification) {
+                logger.warn(`SAS_ACCEPT Participant ${pId} does not have valid sasVerification`);
+                this.eventEmitter.emit(
+                    OlmAdapterEvents.PARTICIPANT_VERIFICATION_COMPLETED,
+                    pId,
+                    false,
+                    E2EEErrors.E2EE_SAS_INVALID_SAS_VERIFICATION);
+
+                return;
+            }
+
+            if (olmData.sasVerification.sasCommitment) {
+                logger.debug(`Already received sas commitment message from ${pId}!`);
+
+                this._sendError(participant, 'Already received sas commitment message from ${pId}!');
```

Comment:
Should we send an error here or just bail with the above log? I know we are a tad inconsistent but this would imply a programming error in the client I think...